packages/react-button/src/components/MenuButton/useMenuButton.ts

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { makeMergePropsCompat, resolveShorthandProps } from '@fluentui/react-utilities';
+import { MenuButtonProps, MenuButtonState } from './MenuButton.types';
+import { useMenuButtonState } from './useMenuButtonState';
+
+/**
+ * Consts listing which props are shorthand props.
+ */
+export const menuButtonShorthandProps = ['children', 'icon', 'menu', 'menuIcon'] as const;
+
+// eslint-disable-next-line deprecation/deprecation
+const mergeProps = makeMergePropsCompat({ deepMerge: menuButtonShorthandProps });
+
+/**
+ * Redefine the component factory, reusing button factory.
+ */
+export const useMenuButton = (props: MenuButtonProps, ref: React.Ref<HTMLElement>, defaultProps?: MenuButtonProps) => {
+  // Ensure that the `ref` prop can be used by other things (like useFocusRects) to refer to the root.
+  // NOTE: We are assuming refs should not mutate to undefined. Either they are passed or not.
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const resolvedRef = ref || React.useRef<HTMLElement>(null);
```

Comment:
Is `resolvedRef` necessary? It doesn't look like you use the value of `state.ref` anywhere other than to forward it to the root element. You could remove `resolvedRef` and just pass the given `ref` directly to the state object.

Or, if you do need to use `state.ref.current` somewhere, you can't rely on the passed ref being a ref object with a `.current` prop (it could be a function), so you'd need to use `useMergedRefs` here instead.

Reply:
You're right! And I'm actually using `useMergedRefs` in `useMenuButtonState` already so I'll remove this part.